doc/Development_Documentation/10_E-Commerce_Framework/15_Payment/10_Recurring_Payments.md

```diff
@@ -56,59 +57,46 @@ public function paymentAction(Request $request)
 
             $sourceOrders[$paymentMethod] = $sourceOrder;
         }
-        $this->view->sourceOrders = $sourceOrders;
+        
+        $templateParams['sourceOrders'] = $sourceOrders;
     }
 
-    $this->view->paymentMethods = $paymentMethods;
+    $templateParams['paymentMethods'] = $paymentMethods;
 
-    // payment.html.php rendered
+    return $this->render('checkout/payment.html.twig', $templateParams);
 }
 ```
 
 #### payment.html.php
 
-```php
+```twig
 ?>
-<form method="post" action="<?= $this->pimcoreUrl(array('action' => 'confirm'), 'checkout', true) ?>">
-
-    <?php if (!empty($this->sourceOrders)): ?>
-
-        <h4><?= $this->t("checkout.use-recurring-payment"); ?></h4>
-
-        <?php
-        foreach ($this->paymentMethods as $paymentMethod) :
-            /* @var \Pimcore\Bundle\EcommerceFrameworkBundle\Model\AbstractOrder $sourceOrder */
-            $sourceOrder = $this->sourceOrders[$paymentMethod];
-            $sourceOrderId = $sourceOrder ? $sourceOrder->getId() : "";
-
-            if (!$sourceOrder) {
-                continue;
-            }
-
-            if ($paymentProvider = $paymentProviderBrick->getPaymentProviderQpay()) :
-                $currentPaymentMethod = $paymentProvider->getAuth_paymentType();
-                ?>
-                <p>
-                    <input name="recurring-payment" value="<?= $sourceOrderId ?>" type="radio">
-                    <strong><?= $currentPaymentMethod ?></strong>
-
-                    <?php
-                    switch ($currentPaymentMethod) {
-                        case "SEPA-DD":
-                            echo $paymentProvider->getAuth_bankAccountOwner() . " " . $paymentProvider->getAuth_bankAccountIBAN();
-                            break;
-                        case "CCARD":
-                            echo $paymentProvider->getAuth_maskedPan() . " " . $paymentProvider->getAuth_expiry();
-                            break;
-                    }
-                    ?>
-                </p>
-                <?
-            endif;
-        endforeach;
-        ?>
+<form method="post" action="{{ pimcore_url({action: 'confirm'}, 'checkout', true) }}">
+
+    {% if sourceOrders is not empty %}
+
+        <h4>{{ 'checkout.use-recurring-payment'|trans }}</h4>
+
+        {% for paymentMethod in paymentMethods if sourceOrders[paymentMethod] is defined %}
```

Comment:
@brusch Is not possible in Twig 3 anymore.

> Adding an if condition on a for tag is deprecated in Twig 2.10. Use a filter filter or an “if” condition inside the “for” body instead (if your condition depends on a variable updated inside the loop).

Reply:
Thanks! Fixed in 61cab12868afb0fe6bd08bbb21486ac9fef9c69b

Reply:
@brusch Sorry for the caused extra work. 🙈